### PR TITLE
Deneb Support

### PR DESCRIPTION
**Pull request by olegshmuelov**

### CODE REVIEW

File: backend/path_sign_proposal_slashing_test.go [hunk 1]
```diff
@@ -189,21 +192,70 @@ func basicSignRequestBlock(blockVersion spec.DataVersion, isBlinded bool, differ
 			Version: spec.DataVersionCapella,
 			Capella: block,
 		}}
-	}
+	// tested against a real block and sig from the goerli testnet (slot 7415521)
```
Comment (alonmuroch) [inline backend/path_sign_proposal_slashing_test.go:195]: where is this taken from?

File: e2e/tests/proposal_reference_signing.go [hunk 2]
```diff
@@ -20,7 +20,8 @@ func referenceBlock(t *testing.T) *spec.VersionedBeaconBlock {
 	blk := &phase0.BeaconBlock{}
 	require.NoError(t, blk.UnmarshalSSZ(blockByts))
 	return &spec.VersionedBeaconBlock{
-		Phase0: blk,
+		Phase0:  blk,
```
Comment (alonmuroch) [inline e2e/tests/proposal_reference_signing.go:23]: we don't test different versions here?

**Review by alonmuroch — CHANGES_REQUESTED** [review]

**Comment by alonmuroch** [comment]
@olegshmuelov from your end we've covered tests for all different scenarios ?

File: go.mod [hunk 3]
```diff
@@ -3,12 +3,12 @@ module github.com/bloxapp/key-vault
 go 1.19
 
 require (
-	github.com/attestantio/go-eth2-client v0.15.2
-	github.com/bloxapp/eth2-key-manager v1.3.1
+	github.com/attestantio/go-eth2-client v0.19.10
+	github.com/bloxapp/eth2-key-manager v1.4.0
```
[the remark this answers is not in the document]
Comment (olegshmuelov) [reply go.mod:7]: the [PR](https://github.com/bloxapp/eth2-key-manager/pull/104) is already merged

File: backend/path_sign_proposal_slashing_test.go (hunk 1 above)
Comment (olegshmuelov) [reply backend/path_sign_proposal_slashing_test.go:195]: The data was taken from the Goerli testnet, specifically from a block signed by a validator whose keys we possess. The slot in question is 7415521. You can reference the corresponding block information through this link: [https://prater.beaconcha.in/block/10389257](https://prater.beaconcha.in/block/10389257).

**Comment by olegshmuelov** [comment]
> @olegshmuelov from your end we've covered tests for all different scenarios ?

Yes, all scenarios tested, including successful attestations, sync committee participation, and crucially, block proposal duties on Sepolia testnet pre/post-fork with this code version.

File: e2e/tests/proposal_reference_signing.go (hunk 2 above)
Comment (olegshmuelov) [reply e2e/tests/proposal_reference_signing.go]: Please see https://github.com/bloxapp/key-vault/pull/100/commits/2eb5a6adb6c6146e817c292224345a0f84dd5834

File: go.mod (hunk 3 above)
[the remark this answers is not in the document]
Comment (alonmuroch) [reply go.mod:7]: added a few questions there

File: backend/path_sign_proposal_slashing_test.go (hunk 1 above)
Comment (alonmuroch) [reply backend/path_sign_proposal_slashing_test.go:195]: please add as comment

**Review by MatheusFranco99 — APPROVED** [review]
LGTM

**Review by moshe-blox — APPROVED** [review]
LGTM

well done 💪

**Review by alonmuroch — APPROVED** [review]

**Pull request merged**